scene/gui/graph_frame.cpp

```diff
@@ -0,0 +1,270 @@
+					sb_frame_texture->set_modulate(tint_color);
+					draw_style_box(sb_frame_texture, body_rect);
+				}
+			} else {
+				draw_style_box(sb_frame_flat, body_rect);
+			}
+
+			// Draw title bar stylebox above.
+			draw_style_box(sb_to_draw_titlebar, titlebar_rect);
+
+			Ref<Texture2D> resizer = get_theme_icon(SNAME("resizer"));
+			Color resizer_color = get_theme_color(SNAME("resizer_color"));
+			if (resizable) {
+				draw_texture(resizer, get_size() - resizer->get_size(), resizer_color);
+			}
+		} break;
+	}
+}
+
+void GraphFrame::_resort() {
+	Ref<StyleBox> sb_frame = get_theme_stylebox(SNAME("frame"));
+	Ref<StyleBox> sb_titlebar = get_theme_stylebox(SNAME("titlebar"));
+
+	// Resort titlebar first.
+	Size2 titlebar_size = Size2(get_size().width, titlebar_hbox->get_size().height);
+	titlebar_size -= sb_titlebar->get_minimum_size();
+	Rect2 titlebar_rect = Rect2(sb_titlebar->get_offset(), titlebar_size);
+
+	fit_child_in_rect(titlebar_hbox, titlebar_rect);
+
+	// After resort the children of the titlebar container may have changed their height (e.g. Label autowrap).
+	Size2i titlebar_min_size = titlebar_hbox->get_combined_minimum_size();
+
+	Size2 size = get_size() - sb_frame->get_minimum_size() - Size2(0, titlebar_min_size.height + sb_titlebar->get_minimum_size().height);
+	Point2 offset = Point2(sb_frame->get_margin(SIDE_LEFT), sb_frame->get_margin(SIDE_TOP) + titlebar_min_size.height + sb_titlebar->get_minimum_size().height);
+
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i, false));
+		if (!c || !c->is_visible_in_tree()) {
+			continue;
+		}
+		if (c->is_set_as_top_level()) {
+			continue;
+		}
+
+		fit_child_in_rect(c, Rect2(offset, size));
+	}
+}
+
+void GraphFrame::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_title", "title"), &GraphFrame::set_title);
+	ClassDB::bind_method(D_METHOD("get_title"), &GraphFrame::get_title);
+
+	ClassDB::bind_method(D_METHOD("get_titlebar_hbox"), &GraphFrame::get_titlebar_hbox);
+
+	ClassDB::bind_method(D_METHOD("set_title_centered", "centered"), &GraphFrame::set_title_centered);
+	ClassDB::bind_method(D_METHOD("is_title_centered"), &GraphFrame::is_title_centered);
+
+	ClassDB::bind_method(D_METHOD("set_drag_margin", "drag_margin"), &GraphFrame::set_drag_margin);
+	ClassDB::bind_method(D_METHOD("get_drag_margin"), &GraphFrame::get_drag_margin);
+
+	ClassDB::bind_method(D_METHOD("set_tint_color_enabled", "p_enable"), &GraphFrame::set_tint_color_enabled);
+	ClassDB::bind_method(D_METHOD("is_tint_color_enabled"), &GraphFrame::is_tint_color_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_tint_color", "p_color"), &GraphFrame::set_tint_color);
+	ClassDB::bind_method(D_METHOD("get_tint_color"), &GraphFrame::get_tint_color);
+
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "title"), "set_title", "get_title");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "title_centered"), "set_title_centered", "is_title_centered");
```

Comment:
As `title` is shared between the two class this could be implemented in the `BaseGraphNode`as it would also benefit `GraphNode` (Wanted this especially while working on my plugin before discovering it was implemented in `GrephFrame`)

Reply:
I didn't implement it in BaseGraphNode since that class should only provide the basic functionality for a node to be used as a GraphEdit child node (dragging/selecting/etc.). A title is needed for GraphNode and GraphFrame, but not necessarily for something like a image backdrop node or a reroute node. I'm still not sure what to do with the title_centered property because its just a convinience property, meaning that you can currently center the title of GraphFrame and GraphNode (and do many more things) with:
(in `_ready()` method)
```
var titlebar_hbox: Container = get_titlebar_hbox()
var label: Label = titlebar_hbox.get_child(0)
label.horizontal_alignment = HORIZONTAL_ALIGNMENT_CENTER
```
Since the method above is much more flexible I am thinking about removing tilte_centered.